packages/cletus/src/operations/dba.tsx

```diff
@@ -736,4 +894,157 @@ const OPERATOR_MAP: Record<string, string> = {
   after: "≥",
   oneOf: " ONE OF ",
   isEmpty: "IS EMPTY",
-};
\ No newline at end of file
+};
+
+/**
+ * Build a description of the type schema for AI extraction
+ */
+function buildTypeSchemaDescription(type: TypeDefinition): string {
+  const fieldDescriptions = type.fields.map(field => {
+    let desc = `- ${field.name} (${field.type})`;
+    if (field.required) desc += ' [required]';
+    if (field.enumOptions) desc += ` [options: ${field.enumOptions.join(', ')}]`;
+    if (field.default !== undefined) desc += ` [default: ${field.default}]`;
+    return desc;
+  }).join('\n');
+  
+  return `Type: ${type.friendlyName}
+${type.description || ''}
+
+Fields:
+${fieldDescriptions}`;
+}
+
+/**
+ * Chunk text with overlap for AI processing
+ */
+function chunkTextWithOverlap(text: string, chunkSize: number, overlap: number): string[] {
+  const chunks: string[] = [];
+  let start = 0;
+  
+  while (start < text.length) {
+    const end = Math.min(start + chunkSize, text.length);
+    chunks.push(text.slice(start, end));
+    
+    if (end >= text.length) break;
+    start = end - overlap;
+  }
+  
+  return chunks;
+}
+
+/**
+ * Extract data records from a text chunk using AI
+ */
+async function extractDataFromChunk(
+  ctx: CletusAIContext,
+  type: TypeDefinition,
+  typeSchema: string,
+  chunk: string
+): Promise<Record<string, any>[]> {
+  const models = ctx.config.getData().user.models;
+  const model = models?.chat;
+  
+  const prompt = `Extract all instances of ${type.friendlyName} from the following text chunk.
+
+${typeSchema}
+
+Return a JSON array of objects, where each object contains the fields defined above. If no instances are found, return an empty array.
+
+Text chunk:
+${chunk}
+
+Respond with ONLY a valid JSON array, no explanation or markdown formatting.`;
+  
+  try {
+    const response = await ctx.ai.chat.get({
+      model,
+      messages: [
+        { role: 'system', content: 'You are a data extraction assistant. Extract structured data from text and respond with valid JSON arrays only.' },
+        { role: 'user', content: prompt },
+      ],
+      maxTokens: 2000,
```

Comment:
Remove Max tokens, and response format and pass zod schema of array of type records. See dba.ts for how it converts type definition into schema.